QudevETH/PycQED_py3
# Enh/split sequences

Changes proposed in this pull request:
- Segment can now be provided resolve_overlap_flag=True which results in the segment resolving overlapping elements by finding the smallest set of mutually overlapping elements, putting all pulses in a new element and resolve all timings, trigger pulses for this new element
- we achieve this by introducing an optional step (resolve_overlap() ) in the resolve_segment() routine after having added all pulses and elements and before the trigger pulses are generated
- this routine first determines which elements are overlapping before combining them into one element using _combine_elements()
- after this step, the trigger pulses are generated for the resolved set of elements taking care of the different hardware requirements

Conducted Tests on virtual setup:
- randomly over lapping sequences get sorted properly into different elements that all mutually overlap
- Rabi sequence works nominally (to be expected)
- Ramsey two cases: a) with spacing where the two pi pulses overlap b) with spacing where they do not overlap. In both cases the routine did resolve the overlap correctly (including trigger pulses)

Next step:
- test the code on BF1 setup
- run Ramsey style sequences as in the virtual case
- make sure no weird timing effects occur (visible in Ramsey if timing is off) 

Potential improvements:
- change resolve_overlap_flag to be a list of AWGs for which this rule should apply.

Once the hardware tests are done I invite @antsr (and either @chellings or @nathlacroix ) for review.
